Sources/SourceControl/GitRepository.swift

```diff
@@ -323,11 +323,21 @@ public class GitRepository: Repository, WorkingCheckout {
 
     // MARK: Working Checkout Interface
 
+    internal func _areAllBranchesPushed() throws -> Bool {
+        let unpushedBranches = try Process.checkNonZeroExit(
+            args: Git.tool, "-C", path.asString, "log", "--branches", "--not", "--remotes").chomp()
+        return unpushedBranches.isEmpty
+    }
+
+    internal func _isHeadOnRemote() throws -> Bool {
```

Comment:
Looks like only `_areAllBranchesPushed` is in the unit tests, so maybe its better to just test the public method `hasUnpushedCommits` and make these methods private or even inline them?